core/irCheck.ml

```diff
@@ -5,30 +5,85 @@ open Ir
 (* ERROR HANDLING*)
 
 let fail_on_ir_type_error = false
-exception IRTypeError of string
+
+(* Artificial "supertype" of all IR types,
+   so we can provide arbitrary IR fragments to the error handling functions *)
+type ir_snippet =
+  | STC    of tail_computation
+  | SVal   of value
+  | SSpec  of special
+  | SBind  of binding
+  | SBinds of binding list
+  | SProg  of program
+  | SNone
+
+let string_of_occurrence : ir_snippet -> string = function
+  | STC tc -> "\noccurring in IR tail computation:\n" ^ Ir.string_of_tail_computation tc
+  | SVal v -> "\noccurring in IR value:\n" ^ Ir.string_of_value v
+  | SSpec s -> "\noccurring in IR special tail computation:\n" ^ Ir.string_of_special s
```

Comment:
Doesn't fit 80 columns.

Reply:
Did we ever decide on using a 80 column limit? I agree that many lines in `irCheck` are too long, but I really hope we're not going for a 80 column limit.

Reply:
No, we never did, so feel free to ignore :-) I personally don't go beyond 80th column.